control_plane/attachment_service/src/service.rs

```diff
@@ -3534,17 +3687,28 @@ impl Service {
         )
     }
 
-    /// Check all tenants for pending reconciliation work, and reconcile those in need
+    /// Check all tenants for pending reconciliation work, and reconcile those in need.
+    /// Additionally, reschedule tenants that require it.
     ///
     /// Returns how many reconciliation tasks were started
     fn reconcile_all(&self) -> usize {
         let mut locked = self.inner.write().unwrap();
-        let pageservers = locked.nodes.clone();
-        locked
-            .tenants
-            .iter_mut()
-            .filter_map(|(_tenant_shard_id, shard)| self.maybe_reconcile_shard(shard, &pageservers))
-            .count()
+        let (nodes, tenants, scheduler) = locked.parts_mut();
+        let pageservers = nodes.clone();
+
+        let mut reconciles_spawned = 0;
+        for (tenant_shard_id, shard) in tenants.iter_mut() {
+            if let Err(err) = shard.maybe_reschedule(&pageservers, scheduler) {
```

Comment:
Let's do this maybe_reschedule part in a later PR -- the logic in `should_schedule` isn't quite right here, as even if a node is unavailable, calling schedule() doesn't do anything unless that node has been bumped with `demote_attached` or similar.